File: java/src/edu/berkeley/cs/scheduling/FiFoScheduler.java
package edu.berkeley.cs.scheduling;

import edu.berkeley.cs.util.LinkedList;

public class FiFoScheduler implements Scheduler {
  protected LinkedList<Job> jobs = new LinkedList<>();

  public FiFoScheduler(int numJobs) {
    for (int i = 0; i < numJobs; i++) {
      jobs.insertEnd(new Job());
    }
  }

  @Override
  public int simulate() {
    // TODO: complete this function
    return 0;
  }
}

File: python/src/scheduling/fifo_scheduler.py
from python.src.scheduling.job import Job
from python.src.scheduling.scheduler import Scheduler
from python.src.util.linkedlist import LinkedList


class FiFoScheduler(Scheduler):

    def __init__(self, num_jobs):
        self.jobs = LinkedList()
        for i in range(num_jobs):
            self.jobs.insert_end(Job())

    def simulate(self):
        # TODO: complete this function
        return 0
